### PR TITLE
expose nc_rc_set, nc_rc_get (via rc_set, rc_get module functions).

### PR DESCRIPTION
**Pull request by jswhit**
addresses PR #1347

### CODE REVIEW

File: src/netCDF4/__init__.pyi [hunk 1]
```diff
@@ -624,6 +625,8 @@ def chartostring(
 ) -> npt.NDArray[np.str_] | npt.NDArray[np.bytes_]: ...
 
 def getlibversion() -> str: ...
+def nc_get(key: str) -> str: ...
+def nc_set(key: str, val: str): ...
```
Comment (headtr1ck) [inline src/netCDF4/__init__.pyi:629]:
```suggestion
def nc_set(key: str, val: str) -> None: ...
```

File: src/netCDF4/__init__.pyi [hunk 2]
```diff
@@ -624,6 +625,8 @@ def chartostring(
 ) -> npt.NDArray[np.str_] | npt.NDArray[np.bytes_]: ...
 
 def getlibversion() -> str: ...
+def nc_get(key: str) -> str: ...
```
Comment (headtr1ck) [inline src/netCDF4/__init__.pyi:628]:
```suggestion
def rc_get(key: str) -> str: ...
```
If I am not mistaken. Same for set.

**Review by headtr1ck — COMMENTED** [review]
You need to import the newly added things in _netCDF4.pyi to fix the stubtest

File: src/netCDF4/__init__.pyi [hunk 3]
```diff
@@ -624,6 +625,8 @@ def chartostring(
 ) -> npt.NDArray[np.str_] | npt.NDArray[np.bytes_]: ...
 
 def getlibversion() -> str: ...
+def rc_get(key: str) -> str: ...
```
Comment (mulimoen) [inline src/netCDF4/__init__.pyi:628]: I think this signature should be
```suggestion
def rc_get(key: str) -> str | None: ...
```
and a null pointer check is necessary

Comment (headtr1ck) [reply src/netCDF4/__init__.pyi:628]: I don't see where it should return None.

Comment (jswhit) [reply src/netCDF4/__init__.pyi]: rc_get now checks for null pointer, returns None

**Pull request merged**